### DISCUSSION TITLE
Parser and Tokenizer error recovery

### DISCUSSION THREAD

**Discussion by m-holger** [qpdf/qpdf-dev#12] [General]
The parser treats unrecognised tokens as strings. What is the reason for this behaviour. Specifically, was this behaviour chosen because it is most likely to correctly repair the pdf file or was it chosen because it is most useful for inspecting a damaged file (or is there some other less obvious reason)?

**Comment by jberkenbilt** [comment]
I don't remember, but my guess would be that it's a type that can hold an arbitrary sequence of bytes, so it seems like the least lossy approach other than perhaps introducing a completely new token type. I suspect that decision must have been made in the very distant past.
